sound/soc/sof/control.c

```diff
@@ -431,24 +432,38 @@ int snd_sof_bytes_ext_put(struct snd_kcontrol *kcontrol,
 		return -EINVAL;
 	}
 
-	if (copy_from_user(cdata->data, tlvd->tlv, header.length))
+	/* Allocate temporary buffer for new cdata */
+	new_cdata = kzalloc(sizeof(*new_cdata) + header.length, GFP_KERNEL);
```

Comment:
Is header.length validated anywhere as this can potentially allocate 4G and run OOM.

Reply:
Good point, I need to find out a check for it.

Reply:
I could find nothing from ALSA/ASoC headers so I defined SND_SOF_CTL_TLV_LENGTH_MAX in sof-priv.h. It's set now to 1Mbyte. It's a lot but should avoid maintaining the value up for now. FW will  anyway error if allocation there exceeds available and fail the IPC cleanly.